nncf/ptq/examples/bert/bert_quantization.py

```diff
@@ -0,0 +1,179 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import os
+import subprocess
+from pathlib import Path
+from typing import Tuple
+
+import openvino
+from openvino.runtime import Core
+import torch
+import datasets
+import evaluate
+import transformers
+import numpy as np
+
+from nncf import ptq
+from nncf.common.utils.logger import logger as nncf_logger
+
+
+FILE = Path(__file__).resolve()
+# Relative path to the `bert` directory.
+ROOT = Path(os.path.relpath(FILE.parent, Path.cwd()))
+# Path to the directory where the original and quantized IR will be saved.
+MODEL_DIR = ROOT.joinpath('bert_quantization')
+# Path to the pre-trained model directory.
+PRETRAINED_MODEL_DIR = ROOT.joinpath('MRPC')
+
+TASK_NAME = 'mrpc'
+MAX_SEQ_LENGTH = 128
+
+
+def run_example():
+    """
+    Runs the BERT quantization example.
+    """
+    # Step 1: Prepare OpenVINO model.
+    ir_model_xml, ir_model_bin = torch_to_openvino_model()
+    ie = Core()
+    original_model = ie.read_model(model=ir_model_xml, weights=ir_model_bin)
+
+    # Step 2: Create dataset.
+    data_source = create_val_dataset()
+
+    # Step 3: Apply quantization algorithm.
+
+    # Define the transformation method. This method should
```

Comment:
Can you please be more specific regarding what is the expected input of the model and output of the dataset class, as to what happens in this function?

Reply:
FIxed.